mlops/bicep/modules/resources/storage_private.bicep

```diff
@@ -0,0 +1,170 @@
+// Creates a storage account, private endpoints and DNS zones
+@description('Azure region of the deployment')
+param location string
+
+@description('Tags to add to the resources')
+param tags object
+
+@description('Name of the storage account')
+param storageName string
+
+@description('Resource ID of the subnets allowed into this storage')
+param subnetIds array
+
+@allowed(['Enabled','vNetOnly','Disabled'])
+@description('Allow or disallow public network access to Storage Account.')
+param publicNetworkAccess string = 'Disabled' // for Disabled, you'd need to create private endpoints
+
+@allowed([
+  'Standard_LRS'
+  'Standard_ZRS'
+  'Standard_GRS'
+  'Standard_GZRS'
+  'Standard_RAGRS'
+  'Standard_RAGZRS'
+  'Premium_LRS'
+  'Premium_ZRS'
+])
+
+@description('Storage SKU')
+param storageSKU string = 'Standard_LRS'
+
+var storageNameCleaned = replace(storageName, '-', '')
+
+// settings depending on publicNetworkAccess
+// no need for subnetIds if publicNetworkAccess is Enabled
+var storageAllowedSubnetIds = publicNetworkAccess == 'Enabled' ? [] : subnetIds
+// if publicNetworkAccess is Enabled, then default action is Allow
+var storagedefaultAction = publicNetworkAccess == 'Enabled' ? 'Allow' : 'Deny'
+// vNetOnly is just a specific case of Enabled
+var storagepublicNetworkAccess = publicNetworkAccess == 'Disabled' ? 'Disabled' : 'Enabled'
+
+resource storage 'Microsoft.Storage/storageAccounts@2022-05-01' = {
+  name: storageNameCleaned
+  location: location
+  tags: tags
+  sku: {
+    name: storageSKU
+  }
+  kind: 'StorageV2'
+  properties: {
+    accessTier: 'Hot'
+
+    // Allow or disallow public access to all blobs or containers in the storage account.
+    allowBlobPublicAccess: false
+
+    // Allow or disallow cross AAD tenant object replication.
+    allowCrossTenantReplication: false
+
+    // Restrict copy to and from Storage Accounts within an AAD tenant or with Private Links to the same VNet.
+    allowedCopyScope: 'PrivateLink'
+
+    // Indicates whether the storage account permits requests to be authorized with the account access key via Shared Key.
+    allowSharedKeyAccess: true
+
+
+    // Network rule set
+    networkAcls: {
+      // Specifies whether traffic is bypassed for Logging/Metrics/AzureServices.
+      // Possible values are any combination of Logging,Metrics,AzureServices
+      // (For example, "Logging, Metrics"), or None to bypass none of those traffics.
+      bypass: 'AzureServices'
+
+      // Specifies the default action of allow or deny when no other rules match.
+      defaultAction: storagedefaultAction
+
+      // Sets the IP ACL rules
+      // ipRules
+
+      // Sets the resource access rules
+      resourceAccessRules: [
+        // {
+        //   resourceId: resourceId('Microsoft.Storage/storageAccounts/blobServices', storageNameCleaned, 'default')
```

Comment:
do we have a reason why we want to keep this big chunk of commented code?

Reply:
Just because I want to keep it to figure out later, I've added a comment inline to explain.